### PR TITLE
feat: added controllers routes and using knex for db migrations and s…

### PR DESCRIPTION
**Pull request by bradleySuira**
This pr include major changes for database setup and endpoints
- added https://knexjs.org as query builder and migrations utility track future database changes
- added separated controllers and routes
- added separated service layer
- update ui api calls and flow

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -1,30 +1,41 @@
...
+
+- Add a new migration, run: `npm run migrate:make [migration_name]`
+
+### Model Changes
+
+-- TODO
+
+### Database Seed
+
+- To seed the database with test data, run: `npm run seed:run`
```
Comment (johannafransn) [inline backend/README.md:34]: Thanks for adding this!

File: frontend/src/services/MoralisService.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import NetworkService from "./NetworkService";
 
 const MoralisService = {
     getLeaderboard: async function (tokenId: string, contractAddress: string) {
-        return NetworkService.getResourceWithAuth("/v1/moralis/leaderboard/" + tokenId + "/" + contractAddress);
+        return NetworkService.getResourceWithAuth("/v1/nfg/leaderboard/moralis/" + tokenId + "/" + contractAddress);
```
Comment (johannafransn) [inline frontend/src/services/MoralisService.ts]: I think this is a typo that will break the leaderboard, `/v1/nfg/leaderboard/moralis/` should be -> `/v1/nft/leaderboard/moralis/`

**Review by johannafransn — APPROVED** [review]
Very well done 👏 thanks for the hard work!

File: frontend/src/services/MoralisService.ts (hunk 2 above)
Comment (bradleySuira) [reply frontend/src/services/MoralisService.ts]: fixed now, thanks!

**Pull request merged**